sdk/eventhub/event-hubs/src/eventHubSender.ts

```diff
@@ -506,6 +503,8 @@ export class EventHubSender extends LinkEntity {
           );
 
           try {
+            this._sender!.sendTimeoutInSeconds =
+              (retryOptions.timeoutInMs! - timeTakenByInit) / 1000;
```

Comment:
It's possible to have negative seconds here, right? Is that intentional? 

At some point we might decide to stop enforcing the "at least 60 seconds" rule.

Reply:
This can only be negative if the `timeTakenByInit` exceeds `retryTimeout`.
That can't happen because we have a settimeout 
`const waitTimer = setTimeout(actionAfterTimeout, retryOptions.timeoutInMs);` that throws an error in case the `retryTimeout` is reached. 

 `timeTakenByInit` exceeding `retryTimeout` would mean that the `settimeout` would have busted already. 

Hence,  `timeTakenByInit` would not exceed `retryTimeout` at line 507.

That being said, would you still like an additional guard/check over here?

Reply:
I think it's worth writing the guard. I thought setTimeout() wasn't guaranteed to execute directly on time (ie, it could be _after_ the timeout is specified for various reasons).

@chradek , you have any thoughts on this? Is the pattern we're using here safe?

Reply:
I'm all for doing the check. If for some reason the time is 0, we know that the send won't come back in time, so why even invoke it? Same for if it is negative...we can't cancel an `init` currently right? So it could be possible that we timeout because the init took too long, but then the init eventually completes and we enter this code path, if I'm reading this right. We should just return at this point too (assuming that the timeout will be or has been triggered).

Reply:
My only input is that if we make the change here, lets ensure we do the change in Service Bus as well